src/table/matrix_table.cpp

```diff
@@ -90,6 +90,19 @@ void MatrixWorkerTable<T>::Get(const std::vector<integer_t>& row_ids,
 }
 
 template <typename T>
+void MatrixWorkerTable<T>::Get(T* data, size_t size, integer_t* row_ids,
```

Comment:
In that function, the size means the size of each array in ```data_vec```. I think your function should be the size of data array, which is  num_col * row_ids_size